library/core/tests/slice.rs

```diff
@@ -1191,6 +1191,27 @@ fn test_rchunks_exact_mut_zip() {
     assert_eq!(v1, [0, 16, 17, 22, 23]);
 }
 
+#[test]
+fn chunks_mut_are_send_and_sync() {
+    use std::cell::Cell;
+    use std::slice::{ChunksExactMut, ChunksMut, RChunksExactMut, RChunksMut};
+    use std::sync::MutexGuard;
+
+    #[allow(unused)]
+    fn assert_send_and_sync()
+    where
+        ChunksMut<'static, Cell<i32>>: Send,
+        ChunksMut<'static, MutexGuard<'static, u32>>: Sync,
+        ChunksExactMut<'static, Cell<i32>>: Send,
+        ChunksExactMut<'static, MutexGuard<'static, u32>>: Sync,
+        RChunksMut<'static, Cell<i32>>: Send,
+        RChunksMut<'static, MutexGuard<'static, u32>>: Sync,
+        RChunksExactMut<'static, Cell<i32>>: Send,
+        RChunksExactMut<'static, MutexGuard<'static, u32>>: Sync,
```

Comment:
Is that a reliable test? Don't we allow `False` where clauses that can never be satisfied (or might allow such things in the future)?

IMO the test should also call this function, to ensure its where clauses can actually be satisfied.

Reply:
Yeah, we need an instantiation of it (not necessarily a call), _à là_:

```rs
let _ =  assert_send_and_sync;
```

I _guess_ a `#[deny(trivial_bounds)]` could do the job as well, since a lint is otherwise fired in these cases

Reply:
I may have time to address this in a follow-up tomorrow. I didn't want to delay this PR, and I don't have any spare time for this in the near future.

Reply:
https://github.com/rust-lang/rust/pull/100112